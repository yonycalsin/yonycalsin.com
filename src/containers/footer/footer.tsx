import Link from 'next/link'

import { Anchor } from 'components'

function Footer() {
  return (
    <footer className="my-6">
      <div className="container">
        <footer className="p-4 bg-white rounded-lg shadow flex flex-col md:flex-row md:items-center md:justify-between md:p-6">
          <span className="text-gray-500">
            © 2022{' '}
            <a href="https://yonycalsin.com" className="hover:underline">
              Yony Calsin™
            </a>
            . All Rights Reserved.
          </span>
          <ul className="flex flex-wrap gap-3 items-center justify-center">
            <li>
              <Link href="/me" legacyBehavior passHref>
                <Anchor>About</Anchor>
              </Link>
            </li>
            <li>
              <Link href="/contact" legacyBehavior passHref>
                <Anchor>Contact</Anchor>
              </Link>
            </li>
          </ul>
        </footer>
      </div>
    </footer>
  )
}

export default Footer
