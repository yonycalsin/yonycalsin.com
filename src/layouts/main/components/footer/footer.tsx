import * as React from 'react'
import { Box, Container, Text, useColorModeValue, VStack } from '@chakra-ui/react'

export function Footer() {
  const backgroundColor = useColorModeValue('gray.50', 'gray.900')

  return (
    <Box as="footer" bg={backgroundColor} py="12">
      <Container maxW="container.md">
        <VStack>
          <Text fontWeight="bold">Created By Yony Calsin</Text>
          <Text textColor="gray.600">Built with Next.js, MDX, Chakra-Ui and Vercel</Text>
        </VStack>
      </Container>
    </Box>
  )
}
