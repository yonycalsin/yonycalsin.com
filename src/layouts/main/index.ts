export * from './components'

// @yonycalsin/eslint-import-sort/typescript-react-exports group
export { default as MainLayout } from './main'
