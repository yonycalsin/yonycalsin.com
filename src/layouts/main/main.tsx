import * as React from 'react'

import type { MainLayoutProps } from 'typings/layouts'
import { Footer, Navbar } from 'containers'
import { FloatSocials } from 'components'

function MainLayout(props: MainLayoutProps) {
  const { children } = props

  return (
    <>
      <Navbar />
      {children}
      <Footer />
      <FloatSocials />
    </>
  )
}

export default MainLayout
