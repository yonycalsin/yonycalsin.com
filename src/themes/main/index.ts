export { default as ThemeMain } from './main'
