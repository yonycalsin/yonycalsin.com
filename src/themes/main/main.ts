import { extendTheme } from '@chakra-ui/react'

import colors from './options/colors'
import config from './options/config'
import fontSizes from './options/font-sizes'
import fonts from './options/fonts'
import mdx from './options/mdx'
import semanticTokens from './options/semantic-tokens'
import shadows from './options/shadows'
import styles from './options/styles'

const ThemeMain = extendTheme({
  config,
  fonts,
  semanticTokens,
  styles,
  fontSizes,
  colors,
  shadows,
  mdx,
})

export default ThemeMain
