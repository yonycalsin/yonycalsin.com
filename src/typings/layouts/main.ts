import type * as React from 'react'

interface MainLayoutProps {
  children: React.ReactNode
}

export type { MainLayoutProps }
