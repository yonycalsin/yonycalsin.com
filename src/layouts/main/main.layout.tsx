import * as React from 'react'
import { Container } from '@chakra-ui/react'

import { FloatSocials } from '~/components'
import { PageTransition } from '~/components/page-transition/page-transition'

import { Footer } from './components/footer/footer'
import { Header } from './components/header/header'

type MainLayoutProps = {
  children: React.ReactNode
}

export const MainLayout = (props: MainLayoutProps) => {
  const { children } = props

  return (
    <>
      <Header />
      <Container maxW="container.md" as="main">
        <PageTransition>{children}</PageTransition>
      </Container>
      <Footer />
      <FloatSocials />
    </>
  )
}
