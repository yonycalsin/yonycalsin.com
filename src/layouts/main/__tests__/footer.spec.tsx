import { ChakraProvider } from '@chakra-ui/react'
import { render } from '@testing-library/react'

import { mainTheme } from '~/themes/main'

import { Footer } from '../components/footer/footer'

describe('Layouts / Main', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    })
  })

  it('renders the layout footer', () => {
    const view = render(
      <ChakraProvider theme={mainTheme}>
        <Footer />
      </ChakraProvider>,
    )

    expect(view.baseElement).toMatchSnapshot()
  })
})
