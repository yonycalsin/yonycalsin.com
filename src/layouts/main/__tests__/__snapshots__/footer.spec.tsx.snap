// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts / Main renders the layout footer 1`] = `
.emotion-0 {
  background: var(--yony-colors-gray-900);
  padding-top: var(--yony-space-12);
  padding-bottom: var(--yony-space-12);
}

.emotion-1 {
  width: 100%;
  -webkit-margin-start: auto;
  margin-inline-start: auto;
  -webkit-margin-end: auto;
  margin-inline-end: auto;
  max-width: var(--yony-sizes-container-md);
  -webkit-padding-start: var(--yony-space-4);
  padding-inline-start: var(--yony-space-4);
  -webkit-padding-end: var(--yony-space-4);
  padding-inline-end: var(--yony-space-4);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-3 {
  font-weight: var(--yony-fontWeights-bold);
}

.emotion-4 {
  color: var(--yony-colors-gray-600);
}

.emotion-5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: baseline;
  outline: 2px solid transparent;
  outline-offset: 2px;
  line-height: var(--yony-lineHeights-normal);
  border-radius: var(--yony-radii-md);
  font-weight: var(--yony-fontWeights-semibold);
  transition-property: var(--yony-transition-property-common);
  transition-duration: var(--yony-transition-duration-normal);
  height: auto;
  min-width: var(--yony-sizes-10);
  font-size: var(--yony-fontSizes-md);
  -webkit-padding-start: var(--yony-space-4);
  padding-inline-start: var(--yony-space-4);
  -webkit-padding-end: var(--yony-space-4);
  padding-inline-end: var(--yony-space-4);
  padding: 0px;
  color: var(--yony-colors-green-200);
}

.emotion-5:focus-visible,
.emotion-5[data-focus-visible] {
  box-shadow: var(--yony-shadows-outline);
}

.emotion-5:disabled,
.emotion-5[disabled],
.emotion-5[aria-disabled=true],
.emotion-5[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--yony-shadows-none);
}

.emotion-5:hover,
.emotion-5[data-hover] {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5:hover:disabled,
.emotion-5[data-hover]:disabled,
.emotion-5:hover[disabled],
.emotion-5[data-hover][disabled],
.emotion-5:hover[aria-disabled=true],
.emotion-5[data-hover][aria-disabled=true],
.emotion-5:hover[data-disabled],
.emotion-5[data-hover][data-disabled] {
  background: initial;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5:active,
.emotion-5[data-active] {
  color: var(--yony-colors-green-500);
}

.emotion-6 {
  font-family: var(--yony-fonts-mono);
  background: whiteAlpha;
  border-radius: var(--yony-radii-md);
  border-width: 1px;
  border-bottom-width: 3px;
  font-size: 0.8em;
  font-weight: var(--yony-fontWeights-bold);
  line-height: var(--yony-lineHeights-normal);
  -webkit-padding-start: 0.4em;
  padding-inline-start: 0.4em;
  -webkit-padding-end: 0.4em;
  padding-inline-end: 0.4em;
  white-space: nowrap;
  -webkit-margin-start: var(--yony-space-2);
  margin-inline-start: var(--yony-space-2);
  -webkit-margin-end: var(--yony-space-2);
  margin-inline-end: var(--yony-space-2);
}

<body
  class="chakra-ui-dark"
>
  <div>
    <footer
      class="emotion-0"
    >
      <div
        class="chakra-container emotion-1"
      >
        <div
          class="chakra-stack emotion-2"
        >
          <p
            class="chakra-text emotion-3"
          >
            Created By Yony Calsin
          </p>
          <p
            class="chakra-text emotion-4"
          >
            Built with Next.js, MDX, Chakra-Ui and Vercel
          </p>
          <button
            class="chakra-button emotion-5"
            type="button"
          >
            Press 
            <kbd
              class="chakra-kbd emotion-6"
            >
              Ctrl
            </kbd>
            +
            <kbd
              class="chakra-kbd emotion-6"
            >
              K
            </kbd>
             to start â†’
          </button>
        </div>
      </div>
    </footer>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>
  <div
    class="chakra-portal"
  >
    <ul
      aria-live="polite"
      id="chakra-toast-manager-top"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      aria-live="polite"
      id="chakra-toast-manager-top-left"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      aria-live="polite"
      id="chakra-toast-manager-top-right"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      aria-live="polite"
      id="chakra-toast-manager-bottom-left"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      aria-live="polite"
      id="chakra-toast-manager-bottom"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      aria-live="polite"
      id="chakra-toast-manager-bottom-right"
      role="region"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
</body>
`;
